checkfield/checkfield.go

```diff
@@ -35,34 +36,62 @@ func CheckRequiredFields(msg interface{}, requiredFields []string) error {
 }
 
 // CheckOutputOnlyFields implements follows https://google.aip.dev/203#output-only
+// TODO Limitation: can't handle struct
 func CheckOutputOnlyFields(msg interface{}, outputOnlyFields []string) error {
 	for _, field := range outputOnlyFields {
 		f := reflect.Indirect(reflect.ValueOf(msg)).FieldByName(field)
 		switch f.Kind() {
-		case reflect.Int32:
+		case reflect.Bool:
+			reflect.ValueOf(msg).Elem().FieldByName(field).SetBool(false)
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 			reflect.ValueOf(msg).Elem().FieldByName(field).SetInt(0)
+		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+			reflect.ValueOf(msg).Elem().FieldByName(field).SetUint(uint64(0))
+		case reflect.Float32, reflect.Float64:
+			reflect.ValueOf(msg).Elem().FieldByName(field).SetFloat(0)
 		case reflect.String:
 			reflect.ValueOf(msg).Elem().FieldByName(field).SetString("")
 		case reflect.Ptr:
 			reflect.ValueOf(msg).Elem().FieldByName(field).Set(reflect.Zero(f.Type()))
-		case reflect.Struct:
-			if err := CheckOutputOnlyFields(f, outputOnlyFields); err != nil {
```

Comment:
The `outputOnlyFields` is for struct `msg`, not for any field in `msg`, so it doesn't make sense to use recursion here.

Reply:
The same issue addressed.